app/controllers/assignments_controller.rb

```diff
@@ -36,28 +36,55 @@ def index
   end
 
   def new
-    @files = Assignment.new
-    @files.course_id = @course.id
-    @files.due_date = (Time.now + 1.week).end_of_day.strftime("%Y/%m/%d %H:%M")
-    @files.available = Time.now.strftime("%Y/%m/%d %H:%M")
-    @files.lateness_config_id = @course.lateness_config_id
-    @files.request_time_taken = true
+    last_assn = @course.assignments.order(created_at: :desc).first
+    if @assignment.is_a? Files
```

Comment:
I don't follow...  Your earlier code had had `@files = Files.new... @exam = Exam.new ... @questions = Questions.new...`, but that discarded the partially-built model if there was an error and the page needed to be reloaded (see the end of `#create`).  So I added the if-is-a logic to reuse the model object with all its partially-filled-in state, as needed.  I'm not sure what single method would be useful here to produce the same result.